### ISSUE TITLE
Support for expressions in attributes + internationalization

### ISSUE THREAD

**Issue by anzorb** [mikeric/rivets#290]
Hi,

First off, I'd like to say rivets is a fantastic library, and I am pleased to the minimalistic approach it takes.

I've been playing around with ways to expose the ability to mark strings for internationalization, which can be passed to a library (such as jsgettext (http://jsgettext.berlios.de/)), any time the model changes.

Angular has a few plugins that do this, I've put together a sample using angular-gettext, which I am using as a baseline (and I feel rivets can do this much better, btw).

https://github.com/anzorb/Angular-gettext-sample

There are two places where strings may need to be translated.
1. DOM node textContent
   
   ``` html
       <p>This string was translated { num } times</p>
   ```
   
   Updating the model, will pass the resulting string to an ii8n library, which will handle the correct pluralization for the selected locale. I was able to expose this in rivets with a few lines of code.
2. Inside of attributes
   
   ``` html
       <input type="text" rv-value="This string was translated { num } times"/>
   ```
   
   rivets does not support expressions inside of attributes, and I'm sure there is a good architectural reason for this, but the way I see it, to support these, things will just have to be reordered, rather than a major rewrite. Let me know your thoughts.

Also, I feel this could be optional, if the string is wrapped with {}, kind of how Angular does this (or even better ;)):

``` html
<input type="text" rv-value="{ num + ' items selected' | formatter }"/>
```

This would retain the formatter capabilities, and provide more powerful features.

If the `<input>` tag makes no sense in this context, looking into the future with custom elements, which heavily rely on attributes, supporting expressions in attributes is important IMHO.

``` html
<x-titlebar rv-data-title="{ num + ' items selected' }"></x-titlebar> 
```

In summary, there are two parts that would need work to support internationalization plugins for rivets.
1. Allow for expressions inside of attributes
2. Provide a convenient way to add a filter attribute so that the final value is passed to a function provided by the developer in the config. This will unify both attributes and node text content, so we can do the following:

``` html
<p translate>This string was translated { num } times</p>
```

``` html
<x-titlebar translate rv-data-title="{ num + ' items selected' }"></x-titlebar> 
```

OR 

``` html
<x-titlebar translate rv-data-title="{ num } items selected"></x-titlebar> 
```

Let me know what you think. I'll be happy if you answer some of these questions and maybe point me to the right direction. So far, I've only seen angular handle such requirements well, but I'd like to have this functionality in Backbone, or any other non-monolithic frameworks + rivets. ii8n is becoming more and more important in the real-world.

Anzor

**Comment by marcospassos** [comment]
Expressions are something really missing in this awesome library.

**Comment by jhnns** [comment]
The problem: expressions are not lightweight at all. Angular's expressions parsing engine [requires 1000 LoC](https://github.com/peerigon/angular-expressions/blob/master/lib/parse.js)...

**Comment by marcospassos** [comment]
I agree, be lightweight is one of the biggest differentials of Rivets. However, we can code it as a decoupled component for optional use. We can also simplify the DSL to make it more lightweight.

I am willing to help code the parser.

**Comment by psibernetic** [comment]
Personally, I feel like building a great set of formatters and binders as an optional Rivets component is a better approach than introducing a huge bug surface area for things that are already supported in another way.
